lib/recent_test.go

```diff
@@ -0,0 +1,89 @@
+package lib
+
+import (
+	"github.com/stretchr/testify/assert"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+func Test_convertData(t *testing.T) {
+	recentFileContent, err := os.ReadFile("../test-data/recent/RECENT_OLD_FORMAT")
+	if err != nil {
+		logger.Error("Could not open file ../test-data/recent/RECENT_OLD_FORMAT")
+	}
+
+	var recentFileData RecentFiles
+	convertData(recentFileContent, &recentFileData)
+	assert.Equal(t, "1.5.6", recentFileData.Terraform[0])
+	assert.Equal(t, "0.13.0-rc1", recentFileData.Terraform[1])
+	assert.Equal(t, "1.0.11", recentFileData.Terraform[2])
+	assert.Equal(t, 3, len(recentFileData.Terraform))
+	assert.Equal(t, 0, len(recentFileData.Tofu))
+}
+
+func Test_saveFile(t *testing.T) {
+	var recentFileData = RecentFiles{
+		Terraform: []string{"1.2.3", "4.5.6"},
+		Tofu:      []string{"6.6.6"},
+	}
+	temp, err := os.MkdirTemp("", "recent-test")
+	defer func(path string) {
+		_ = os.RemoveAll(path)
+	}(temp)
+	if err != nil {
```

Comment:
The error should be checked before performing the defer, I think

Reply:
Done